core/src/main/scala/org/apache/pekko/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -50,49 +54,33 @@ class JournalQueries(
     }
   }
 
-  private def selectAllJournalForPersistenceIdDesc(persistenceId: Rep[String]) =
-    selectAllJournalForPersistenceId(persistenceId).sortBy(_.sequenceNumber.desc)
-
   private def selectAllJournalForPersistenceId(persistenceId: Rep[String]) =
     JournalTable.filter(_.persistenceId === persistenceId).sortBy(_.sequenceNumber.desc)
 
   def delete(persistenceId: String, toSequenceNr: Long) = {
     JournalTable.filter(_.persistenceId === persistenceId).filter(_.sequenceNumber <= toSequenceNr).delete
   }
 
-  def markJournalMessagesAsDeleted(persistenceId: String, maxSequenceNr: Long) =
+  def markSeqNrJournalMessagesAsDeleted(persistenceId: String, sequenceNr: Long) =
     JournalTable
       .filter(_.persistenceId === persistenceId)
-      .filter(_.sequenceNumber <= maxSequenceNr)
+      .filter(_.sequenceNumber === sequenceNr)
       .filter(_.deleted === false)
       .map(_.deleted)
       .update(true)
 
   private def _highestSequenceNrForPersistenceId(persistenceId: Rep[String]): Rep[Option[Long]] =
     selectAllJournalForPersistenceId(persistenceId).take(1).map(_.sequenceNumber).max
 
-  private def _highestMarkedSequenceNrForPersistenceId(persistenceId: Rep[String]): Rep[Option[Long]] =
-    selectAllJournalForPersistenceId(persistenceId).filter(_.deleted === true).take(1).map(_.sequenceNumber).max
-
-  val highestSequenceNrForPersistenceId = Compiled(_highestSequenceNrForPersistenceId _)
-
-  val highestMarkedSequenceNrForPersistenceId = Compiled(_highestMarkedSequenceNrForPersistenceId _)
-
-  private def _selectByPersistenceIdAndMaxSequenceNumber(persistenceId: Rep[String], maxSequenceNr: Rep[Long]) =
-    selectAllJournalForPersistenceIdDesc(persistenceId).filter(_.sequenceNumber <= maxSequenceNr)
-
-  val selectByPersistenceIdAndMaxSequenceNumber = Compiled(_selectByPersistenceIdAndMaxSequenceNumber _)
-
-  private def _allPersistenceIdsDistinct: Query[Rep[String], String, Seq] =
-    JournalTable.map(_.persistenceId).distinct
-
-  val allPersistenceIdsDistinct = Compiled(_allPersistenceIdsDistinct)
-
-  def journalRowByPersistenceIds(persistenceIds: Iterable[String]): Query[Rep[String], String, Seq] =
-    for {
-      query <- JournalTable.map(_.persistenceId)
-      if query.inSetBind(persistenceIds)
-    } yield query
+  private def _highestSequenceNrForPersistenceIdBefore(
+      persistenceId: Rep[String],
+      maxSequenceNr: Rep[Long]): Rep[Long] =
+    selectAllJournalForPersistenceId(persistenceId)
+      .filter(_.sequenceNumber <= maxSequenceNr)
+      .take(1)
```

Comment:
not sure this will be safe with a `take(1)` and then `max`. I expected something like `filter order by limit 1` or `filter ... max`

Reply:
ordering in the code is not required, they will be compiled to SQL in the end.

Reply:
this had been fixed.